spec/lib/guardian_spec.rb

```diff
@@ -369,11 +364,10 @@
       group.add(user)
       expect(Guardian.new(user).can_send_private_message?(group)).to eq(true)
 
-      SiteSetting.enable_personal_messages = false
       SiteSetting.personal_message_enabled_groups = Group::AUTO_GROUPS[:trust_level_0]
       expect(Guardian.new(trust_level_0).can_send_private_message?(group)).to eq(false)
 
-      #  group membership trumps min_trust_to_send_messages setting
+      # group membership trumps personal_message_enabled_groups setting
```

Comment:
Does this updated comment make sense? 🤔

Reply:
Actually yeah this is no longer applicable since this old condition is gone now:

```ruby
# Have to be a basic level at least
    (is_group || @user.has_trust_level?(SiteSetting.min_trust_to_send_messageshttps://github.com/discourse/discourse/commit/c6cfbebf1f745de349ba32ddcb1844838cede33f || notify_moderators) &&
```

The user now needs to be in `personal_message_enabled_groups` as well as being part of the group membership based on the messageable settings:

```ruby
(from_system || is_user || Group.messageable(@user).where(id: target.id).exists? || notify_moderators) &&
```

Updated spec to reflect this, and fixed other failing core specs.